secretsstoresynccontroller/api/v1alpha1/secretstoresync_types.go

```diff
@@ -0,0 +1,144 @@
+	// SecretDataValueSource is the data source value of the secret defined in the Secret Provider Class.
+	// +kubebuilder:validation:Required
+	SecretDataValueSource string `json:"secretDataValueSource"`
+
+	// SecretDataKey is the key in the Kubernetes secret's data field as described in the Kubernetes API reference:
+	// https://kubernetes.io/docs/reference/kubernetes-api/config-and-storage-resources/secret-v1/
+	// +kubebuilder:validation:Required
+	SecretDataKey string `json:"secretDataKey"`
+}
+
+// SecretObject defines the desired state of synchronized Kubernetes secret objects.
+type SecretObject struct {
+	// Type specifies the type of the Kubernetes secret object, e.g., Opaque. The controller doesn't have permissions
+	// to create a secret object with other types than the ones specified in the Enum.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum="Opaque";"kubernetes.io/basic-auth";"kubernetes.io/ssh-auth";"kubernetes.io/tls"
+	Type string `json:"type"`
+
+	// Data is a slice of SecretObjectData containing secret data source from the Secret Provider Class and the
+	// corresponding data field key used in the Kubernetes secret object.
+	// +kubebuilder:validation:Required
+	Data []SecretObjectData `json:"data"`
+
+	// Labels contains key-value pairs representing labels associated with the Kubernetes secret object.
+	// The labels are used to identify the secret object.
+	// On secret creation, the following label is added: created-by:secrets-store.sync.x-k8s.io.
+	// Creation fails if the label is specified in the SecretSync object with a different value.
+	// On secret update, if the validation admission policy is set, the controller will check if the label
+	// created-by:secrets-store.sync.x-k8s.io is present. If the label is not present, controller fails to
+	// update the secret.
+	// +optional
+	// +kubebuilder:default:={created-by:secrets-store.sync.x-k8s.io}
+	Labels map[string]string `json:"labels,omitempty"`
+
+	// Annotations contains key-value pairs representing annotations associated with the Kubernetes secret object.
+	// On secret creation, the following annotation is added: created-by:secrets-store.sync.x-k8s.io.
+	// Creation fails if the label is specified in the SecretsStore object with a different value.
+	// +optional
+	// +kubebuilder:default:={created-by:secrets-store.sync.x-k8s.io}
+	Annotations map[string]string `json:"annotations,omitempty"`
+}
+
+// SecretStoreSyncSpec defines the desired state for synchronizing secret.
+type SecretStoreSyncSpec struct {
+	// SecretProviderClassName specifies the name of the secret provider class used to pass information to
+	// access the secret store.
+	// +kubebuilder:validation:Required
+	SecretProviderClassName string `json:"secretProviderClassName"`
+
+	// ServiceAccountName specifies the name of the service account used to access the secret store.
+	// +kubebuilder:validation:Required
+	ServiceAccountName string `json:"serviceAccountName"`
```

Comment:
Would be good to have kubebuilder validation on these fields to prevent nonsensical names like `a/b` for service account name.

Reply:
Right I am asking for us to avoid letting the user make a choice that will never work:

```go
// ValidateServiceAccountName can be used to check whether the given service account name is valid.
// Prefix indicates this name will be used as part of generation, in which case
// trailing dashes are allowed.
var ValidateServiceAccountName = NameIsDNSSubdomain
```